### PR TITLE
Pic 2481

### PR DESCRIPTION
**Pull request by vb-moj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/TelemetryService.java [hunk 1]
```diff
@@ -191,6 +191,40 @@ public void trackOffenderProbationStatusUpdateEvent(OffenderEntity offenderEntit
...
+
+        ofNullable(offenderEntity)
+                .map(OffenderEntity::isPreSentenceActivity)
+                .ifPresent((preSentenceActivity) -> properties.put("preSentenceActivity", preSentenceActivity.toString()));
+
+        ofNullable(offenderEntity)
+                .map(OffenderEntity::getAwaitingPsr)
+                .ifPresent((awaitingPsr) -> properties.put("awaitingPsr", awaitingPsr.toString()));
+
+        addRequestProperties(properties);
```
Comment (vb-moj) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/service/TelemetryService.java:223]: Not sure we need to pass all the fields. May be just the CRN is enough?

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/model/ProbationStatusDetailComparator.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package uk.gov.justice.probation.courtcaseservice.service.model;
+
+import java.util.Comparator;
+
+import static java.util.Comparator.*;
+
+public class ProbationStatusDetailComparator {
```
Comment (rschallapalli) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/service/model/ProbationStatusDetailComparator.java:7]: This looks much elegant using the language provision? 
public class ProbationStatusDetailComparator implements Comparator< ProbationStatusDetail> {
 ...
}

**Review by rschallapalli — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/model/ProbationStatusDetailComparator.java (hunk 2 above)
Comment (vb-moj) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/service/model/ProbationStatusDetailComparator.java]: Thanks,  On second thought, don't think we need a comparator at all.

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/TelemetryService.java [hunk 3]
```diff
@@ -191,6 +191,40 @@ public void trackOffenderProbationStatusUpdateEvent(OffenderEntity offenderEntit
         telemetryClient.trackEvent(TelemetryEventType.OFFENDER_PROBATION_STATUS_UPDATED.eventName, properties, Collections.emptyMap());
     }
 
+
+    public void trackOffenderProbationStatusNotUpdateEvent(OffenderEntity offenderEntity) {
```
Comment (rschallapalli) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/service/TelemetryService.java:195]: AC doesn't mention this. Are these helpful? There could be thousands of these in a day. @jhevans

Comment (vb-moj) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/service/TelemetryService.java:195]: This is in the AC. `Probation status update which does not cause change does not trigger database call (log an event to indicate this)`

**Review by rschallapalli — APPROVED** [review]

**Pull request merged**